modules/kotest/src/main/kotlin/org/testcontainers/containers/StartablePerSpecListener.kt

```diff
@@ -0,0 +1,30 @@
+package org.testcontainers.containers
+
+import io.kotest.core.listeners.TestListener
+import io.kotest.core.spec.Spec
+import org.testcontainers.lifecycle.Startable
+
+
+/**
+ * [StartablePerSpecListener] starts the given [startables] before execution of any test in the spec
+ * and stops after execution of all tests.
+ *
+ * [startables] can any of [GenericContainer] [ContainerisedDockerCompose] [LocalStackContainer] etc.
+ *
+ * This should be use when you want to a single container for all test in a single test class.
+ *
+ * @see
+ * [StartablePerTestListener]
+ * */
+
+class StartablePerSpecListener(private vararg val startables: Startable) : TestListener {
```

Comment:
`TestLifecycleAware` support is missing

Reply:
@bsideup  could please explain to me what exactly I need to do, I did not understand it

Reply:
you need to add support for `TestLifecycleAware`, the same way it is done in other test framework integrations.

Reply:
Done